libs/components/phone-field/src/lib/modules/phone-field/phone-field.component.ts

```diff
@@ -402,47 +402,64 @@ export class SkyPhoneFieldComponent implements OnDestroy, OnInit {
     this.#changeDetector.markForCheck();
   }
 
-  public setCountryByDialCode(phoneNumber: string): boolean {
-    if (
-      !phoneNumber ||
-      !phoneNumber.startsWith('+') ||
-      (this.selectedCountry?.dialCode &&
-        phoneNumber.startsWith(`${this.selectedCountry?.dialCode}`))
-    ) {
+  public setCountryByDialCode(phoneNumberRaw: string): boolean {
+    if (!phoneNumberRaw || !phoneNumberRaw.startsWith('+')) {
       return false;
     }
 
+    let dialCode = '';
+    let foundCountry: SkyPhoneFieldCountry | undefined;
     let newCountry: SkyPhoneFieldCountry | undefined;
 
-    for (let i = 1; i < this.#longestDialCodeLength + 1; i++) {
-      const dialCode = phoneNumber.substring(0, i);
-
-      if (this.#defaultCountryData?.dialCode === dialCode) {
-        newCountry = this.#defaultCountryData;
-      } else if (this.selectedCountry?.dialCode !== dialCode) {
-        let foundCountry = this.countries.find(
-          (country) => country.dialCode === dialCode && country.priority === 0,
-        );
-
-        if (foundCountry) {
-          // Ensure that the country that was found is one of the supported countries
-          if (
-            this.supportedCountryISOs &&
-            this.supportedCountryISOs.findIndex(
-              (isoCode) =>
-                isoCode.toUpperCase() === foundCountry?.iso2.toUpperCase(),
-            ) < 0
-          ) {
-            foundCountry = undefined;
-          }
+    try {
+      const phoneNumberParsed =
+        this.#phoneUtils.parseAndKeepRawInput(phoneNumberRaw);
+      const regionCode =
+        this.#phoneUtils.getRegionCodeForNumber(phoneNumberParsed);
+      foundCountry = this.countries.find(
+        (country) => country.iso2.toUpperCase() === regionCode?.toUpperCase(),
+      );
 
-          if (foundCountry !== this.selectedCountry) {
-            newCountry = foundCountry;
+      if (foundCountry && !this.#validateSupportedCountry(foundCountry)) {
+        foundCountry = undefined;
+      }
+    } catch (_) {
+      if (
+        !(
+          this.selectedCountry?.dialCode &&
+          phoneNumberRaw.startsWith(`${this.selectedCountry?.dialCode}`)
+        )
```

Comment:
Can you simplify this for readability?
```typescript
!this.selectedCountry?.dialCode ||
!phoneNumberRaw.startsWith(`${this.selectedCountry?.dialCode}`)
```

Reply:
Updated. I also reverted the playground change as that wasn't intended to be checked in.